packages/medusa/src/api-v2/admin/products/validators.ts

```diff
@@ -202,6 +202,7 @@ export const AdminCreateProduct = z
     tags: z.array(AdminUpdateProductTag).optional(),
     options: z.array(AdminCreateProductOption).optional(),
     variants: z.array(AdminCreateProductVariant).optional(),
+    sales_channels: z.array(z.object({ id: z.string() })).optional(),
```

Comment:
question: Should we allow string IDs too? I know this would be diverging from 1.0, but I don't see the benefit of constructing objects in the client.

Reply:
I do agree, this is due to the legacy usage of providing ids, the naming was always referring to the target object. In other places we do allow the ids too instead of objects

Reply:
@olivermrbl I tried not to introduce too many breaking changes in the API, but we can revisit once we have things working